Framework/DataHandling/src/MaskDetectors.cpp

```diff
@@ -191,10 +239,92 @@ void MaskDetectors::exec() {
   /*
   This rebuild request call, gives the workspace the opportunity to rebuild the
   nearest neighbours map
-  and therfore pick up any detectors newly masked with this algorithm.
+  and therefore pick up any detectors newly masked with this algorithm.
   */
   WS->rebuildNearestNeighbours();
 }
+/* Verifies input ranges are defined and returns these ranges if they are.
+*
+* @return tuple containing min/max ranges provided to algorithm
+*        (from 0 to max histogram range) and Boolean value,
+*        containing true if the ranges are
+*        constrained (or defined in other words)
+*        and false if they are not.
+*/
+std::tuple<size_t, size_t, bool>
+MaskDetectors::getRanges(const MatrixWorkspace_sptr &targWS) {
+  int endIndex = getProperty("EndWorkspaceIndex");
+  int startIndex = getProperty("StartWorkspaceIndex");
+  size_t max_ind = targWS->getNumberHistograms() - 1;
+
+  if (endIndex == EMPTY_INT() && startIndex == 0) {
+    return std::tuple<size_t, size_t, bool>(0, max_ind, false);
+  } else {
+    if (startIndex < 0) {
```

Comment:
*StartWorkspaceIndex* (and *EndWorkspaceIndex*) property has the requirement of being positive by the `mustBePosInt` validator so this check is not necessarily needed.